### PR TITLE
ℤ-Functors

### PR DESCRIPTION
**Pull request by mzeuner**
This PR introduces the category of ℤ-functors, i.e. functors from commutative rings to sets (of a given size), i.e. presheaves on `CommRing ^op`. We show/introduce two things

1. The Yoneda embedding has something that looks like a left adjoint, although not quite, because we run into a mismatch of universe levels.
2. To each ℤ-functor, we can associate the lattice of compact open subfunctors that are classified by the Zariski lattice. This can be used to define the notion of an affine cover.

There are still a bunch of TODOs in `Categories.Instances.ZFunctors` that will hopefully be proved in later PRs. 

Once we have sites (https://github.com/agda/cubical/pull/1031), we can define the Zariski topology on ℤ-functors and define quasi-compact, quasi-separated schemes as the Zariski sheaves that have an affine cover in the sense of this PR. See also the discussion in https://github.com/agda/cubical/issues/657.

The ultimate goal is to do all of that for finitely presented algebras, where we can avoid the size issues. This would give us a predicative and constructive account of schemes of finite presentation (over some base `Spec(A)`).

### CODE REVIEW

File: Cubical/Algebra/DistLattice/Properties.agda [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+  idCompDistLatticeHom : {L M : DistLattice ℓ} (f : DistLatticeHom L M)
+                    → compDistLatticeHom _ _ _ f (idDistLatticeHom M) ≡ f
+  idCompDistLatticeHom = idCompLatticeHom
+
+  compAssocDistLatticeHom : {L M N U : DistLattice ℓ}
+                         (f : DistLatticeHom L M) (g : DistLatticeHom M N) (h : DistLatticeHom N U)
+                       → compDistLatticeHom _ _ _ (compDistLatticeHom _ _ _ f g) h
+                       ≡ compDistLatticeHom _ _ _ f (compDistLatticeHom _ _ _ g h)
+  compAssocDistLatticeHom = compAssocLatticeHom
```
Comment (mortberg) [inline Cubical/Algebra/DistLattice/Properties.agda:55]: Please package this up as a category instance

File: Cubical/Categories/Instances/ZFunctors.agda [hunk 2]
```diff
@@ -0,0 +1,412 @@
...
+    (λ _ _ _ → makeNatTransPath (funExt₂
+                 (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧lAssoc _ _ _)))
+    (λ _ → makeNatTransPath (funExt₂ (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧lRid _)))
+    (λ _ _ → makeNatTransPath (funExt₂ (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧lComm _ _)))
+    (λ _ _ → makeNatTransPath (funExt₂ -- don't know why ∧lAbsorb∨l doesn't work
+               (λ A _ → ZariskiLattice A .snd .DistLatticeStr.absorb _ _ .snd)))
+    (λ _ _ _ → makeNatTransPath (funExt₂ -- same here
+                 (λ A _ → ZariskiLattice A .snd .DistLatticeStr.∧l-dist-∨l _ _ _ .fst)))
+
+  -- TODO: (contravariant) action on morphisms
```
Comment (mortberg) [inline Cubical/Categories/Instances/ZFunctors.agda:394]: Please fix this as it should be easy, then we can merge

**Review by mortberg — CHANGES_REQUESTED** [review]
Good stuff, a few comments that could be fixed before we merge and then you can do the other TODOs as separate PRs

File: Cubical/Algebra/DistLattice/Properties.agda (hunk 1 above)
Comment (mzeuner) [reply Cubical/Algebra/DistLattice/Properties.agda:55]: Done

File: Cubical/Categories/Instances/ZFunctors.agda (hunk 2 above)
Comment (mzeuner) [reply Cubical/Categories/Instances/ZFunctors.agda]: Done

**Review by mortberg — APPROVED** [review]
Nice! Will merge once CI is done

**Pull request merged**